app/controllers/sessions_controller.rb

```diff
@@ -67,6 +66,14 @@ def login_failure(message)
     render template: "sessions/new", status: :unauthorized
   end
 
+  def mfa_recommended_redirect
+    if current_user.mfa_disabled?
+      redirect_to_mfa_setup(notice: t("multifactor_auths.setup_recommended"))
+    else
```

Comment:
We'll need to be more explicit about the `else` case here. Right now, I believe users with mfa configurations set to `ui_and_api` or `ui_and_gem_signin` will also be captured in this else case. So they'll be redirected too. 

```ruby
if current_user.mfa_disabled?
  # etc.
elsif current_user.mfa_ui_only?
  # etc.
end
```

Reply:
That's a good point to raise that I was struggling with. As a part of the [`mfa_recommended?`](https://github.com/Shopify/rubygems.org/pull/15/files#diff-cdec550eeeb8fc63be2b5687170f594651a8d0b7f0465c9d807baef392639b6eR54) check, we look at the user's mfa level. If it's `ui_and_api` or `ui_and_gem_signin`, `mfa_recommended?` would return false and `mfa_recommended_redirect` won't be called in the first place. I figured that it would be dead code to check it again here.

Reply:
Oh I see what you mean now 👍  we can make it more explicit but I think we need to have a default case like redirecting to the dashboard?

Reply:
For anyone that is interested... Betty and I had a quick chat about this and we agreed that this new extracted method did have a smell. We decided to separate the logic in this design as we needed to check the mfa_recommended alongside another conditional:
(e.g. mfa_recommended && other_conditional)

```
return redirect_to_mfa_setup(notice: t("multifactor_auths.setup_recommended")) if mfa_disabled_on_recommended_account
return redirect_to_settings(notice: t("multifactor_auths.strong_mfa_level_recommended")) if weak_mfa_on_recommended_account

def mfa_disabled_on_recommended_account
  current_user.mfa_recommended? && current_user.mfa_disabled?
end

def weak_mfa_on_recommended_account
  current_user.mfa_recommended? && current_user.mfa_ui_only?
end
```

We had trouble naming the concept `mfa_disabled_on_recommended_account` and `weak_mfa_on_recommended_account`, so then we decided the original logic worked well.

Original
<img width="1107" alt="Screen Shot 2022-03-14 at 5 29 26 PM" src="https://user-images.githubusercontent.com/42748004/158271355-cc5e6709-1c43-4f7d-81fa-193fed5a93ae.png">

to
<img width="968" alt="Screen Shot 2022-03-14 at 5 32 43 PM" src="https://user-images.githubusercontent.com/42748004/158271344-e590efe6-716f-4f4f-86cd-befd8d4ab613.png">